test/integration/dss/test_dss_api_file_limit.py

```diff
@@ -0,0 +1,32 @@
+import os
+import resource
+import unittest
+
+from hca.dss import DSSClient
+from hca.dss.util import iter_paths, object_name_builder
+from test import TEST_DIR
+
+
+class TestDssApiFileLimit(unittest.TestCase):
+    staging_bucket = "org-humancellatlas-dss-cli-test"
+    old_open_file_limits = resource.getrlimit(resource.RLIMIT_NOFILE)
+
+    def test_python_open_file_limit_upload(self):
+        bundle_path = os.path.join(TEST_DIR, "upload", "many-files")
+        uploaded_paths = [x.path for x in iter_paths(str(bundle_path))]
+        resource.setrlimit(resource.RLIMIT_NOFILE, (9, self.old_open_file_limits[1]))
+        uploaded_files = [object_name_builder(p, bundle_path) for p in uploaded_paths]
+        client = DSSClient(swagger_url="https://dss.dev.data.humancellatlas.org/v1/swagger.json")
+        manifest = client.upload(src_dir=bundle_path,
+                                 replica="aws",
+                                 staging_bucket=self.staging_bucket)
+        manifest_files = manifest['files']
+        self.assertEqual(list(file['name'] for file in manifest_files).sort(), uploaded_files.sort())
+
+    @classmethod
+    def tearDownClass(cls):
```

Comment:
teardown and setup should be symmetric. Either convert this to a setUp() and tearDown() pair—I don't see why this fixture needs to have class scope—or just use a regular `try:` `finally:` construct inside the test itself. If you do that, consider moving this test method to a test case that already has tests for the upload functionality or move the test case to an existing test module with upload functionality if this needs to be its own test case for some reason (I don't currently see why it would.